docs/adrs/2023-05-08-exposing-metrics.md

```diff
@@ -0,0 +1,179 @@
+# Exposing metrics
+
+Date: 2023-05-08
+
+**Status**: Proposed
+
+## Context
+
+Prometheus metrics are a common way to monitor the cluster. Providing metrics
+can be a helpful way to monitor scale sets and the health of the ephemeral runners.
+
+## Proposal
+
+Two main components are driving the behavior of the scale set:
+
+1. ARC controllers responsible for managing Kubernetes resources.
+2. The `AutoscalingListener`, driver of the autoscaling solution responsible for
+   describing the desired state.
+
+We can approach publishing those metrics in 3 different ways
+
+### Expose Service Monitor, Services for listeners and a Service for the manager
+
+To expose metrics, we would need to create 3 additional resources:
+
+1. `ServiceMonitor` - a resource used by Prometheus to match namespaces and
+   services from where it needs to gather metrics
+2. `Service` for the `gha-runner-scale-set-controller` - service that will
+   target ARC controller `Deployment`
+3. `Service` for each `gha-runner-scale-set` listener - service that will target
+   a single listener pod for each `AutoscalingRunnerSet`
+
+#### Pros
+
+- Easy to control which scale set exposes metrics and which does not.
+- Easy to implement using helm charts in case they are enabled per chart
+  installation.
+
+#### Cons
+
+- With a cluster running many scale sets, we are going to create a lot of
+  resources.
+- In case metrics are enabled on the controller manager level, and they should
+  be applied across all `AutoscalingRunnerSets`, it is difficult to inherit this
+  configuration by applying helm charts.
+
+### Create an aggregator service
```

Comment:
I am personally highly in favour of this approach, especially since this is the same way it was implemented in the legacy modes